src/Algebra/Properties/CancellativeCommutativeSemiring.agda

```diff
@@ -0,0 +1,72 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some theory for CancellativeCommutativeSemiring.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra using (CancellativeCommutativeSemiring)
+import Algebra.GCD
+import Algebra.Primality
+import Algebra.Properties.Semigroup.Divisibility as SemigroupDiv
+import Algebra.Properties.Semiring.Divisibility as SemiringDiv
+open import Data.Product using (_×_; _,_; proj₁; proj₂)
+open import Data.Sum using (_⊎_; inj₁; inj₂)
+open import Function using (case_of_; flip)
+open import Level using (_⊔_)
+open import Relation.Binary using (Rel; Symmetric; Setoid)
+open import Relation.Unary using (Pred)
+
+module Algebra.Properties.CancellativeCommutativeSemiring
+  {α ℓ} (R : CancellativeCommutativeSemiring α ℓ)
+  (open CancellativeCommutativeSemiring R renaming (Carrier to A))
+  where
+
+open SemigroupDiv *-semigroup using (_∣_; _∤_; ∣-trans)
+open SemiringDiv semiring using (∣nonzero⇒≉0)
+open Algebra.Primality _≈_ _*_ 0# 1# using (IsIrreducible; Coprime)
+open Algebra.GCD _≈_ _*_ using (GCD)
+
+0∤1 : 0# ≉ 1# → 0# ∤ 1#
+0∤1 0≉1 (q , q*0≈1) = 0≉1 (trans (sym (zeroʳ q)) q*0≈1)
+
+------------------------------------------------------------------------------
+-- Properties of Irreducibilty, Coprime.
+
+irreducible≉0 : 0# ≉ 1# → (∀ {p} → IsIrreducible p → p ≉ 0#)
```

Comment:
I remove these parentheses.

About ``a, b`` vs ``x, y``.
--------------------------------
In textbooks, prime (or irreducible) elements are usually denoted `p`, `q`.
This a mnemonic carried by a variable name.

Also ``x, y, z`` usually denote more local values that are quantified and may be are often changed. 
When there are the environment parameters and there are local values ``x, y, z``, where the former are stable,
and the latter are quantified inside,
it is usulally written ``a, b, c, p, q ...`` for the environment parameters
and ``x, y, z`` for local values.
For example, in the body of ``record GCD (a b : A)``,
there may be foralled ``x, y`` to expressed the properties of ``GCD a b``.

These denotations make it easier to follow the discourse with formulas.

Please, consider this and announce the decision.

Reply:
Well leaving aside the fact there is an implicit quantifier over `a` and `b` as well, maybe the best way to think about the standard library is that the library itself is a textbook. It too has it's own conventions. You wouldn't want to read a textbook that switched notation on every page right? In the same way, we don't want the library to switch the notation used in every module.

There might be some valid argument for using `p` and `q` for primes, but in the library the names `a`, `b` are reserved for universe levels. Therefore they are not appropriate to use as elements in an algebra.

Reply:
So,  `a` and `b ` are reserved for Level
(probably, changing them will break backwards compatibility). 
I have set this in the commit.

But I understand your sentence about `p` and `q` so that you allow them in the definitions  of
Irreducible and Prime.

> maybe the best way to think about the standard library is that the library itself is a textbook.

It is natural for the _algebra part_ of the library to preserve
_some_ traditions of the algebra textbooks, just not to neglect everything in a rough way.

This is for the following reason.
a) Among the users of this algebra part 99% are mathematicians by education
(if you ask why, I would explain).
And they are used to follow traditions. For example, formally, the sentence
"Let x be any _group_ and y any its subgroup."  is all right. But this will cause a disgust,
unless these denotations are explained by  a very particular reason.
Rather people expect expressions like
"Let G (A, B ...) be any group and H (G', A' ...) any its subgroup."
These traditions also have a certain sense in them, mathematical denotations have a five century experience.
We only need not to neglect everything in without a technical reason.
If we have an opportunity to denote ``(G : Group _ _)``, then all right.
If some technical reasons require (like in Haskell) a small first letter,
OK, write, say, ``(gG : Group _ _)``. And the last and the worst solution will be, say, 
``(x1 : Group _ _)``, not a great tragedy, but just unpleasant.